dali/core/CMakeLists.txt

```diff
@@ -14,6 +14,8 @@
 
 project(dali_core CUDA CXX C)
 
+add_subdirectory(mm)
+
```

Comment:
I see very little gain in making it longer. It will be widely used, so the name _will_ be familiar. I agree that naming some variable somewhere "mm" is not a good idea, because that would need to be discovered case-by-case.

Reply:
I guess we have two separate things here. First is the directory structure, I believe that this could be little bit more verbose, than the namespace (thus I suggested `memory`). As for namespaces name, `mem` look really good. It's only one tiny letter longer than `mm` and gives you huge difference in actual meaning. I'd imagine e.g. reading debug stack trace a lot easier when having `mem` instead of `mm`. Especially if somebody not that accustomed to DALI would need to do it, e.g. as a bug description

Reply:
I still have reservations against `mem` and `memory` because (unlike mm) they drop the second M - the manager/management part. And I definitely don't want something as long as `memory_manager` or as awkard as `memm`.
When we make it "memory" or "mem" it immediately reminds of `#include <memory>` which contains very different things: smart pointers, typed allocators, pointer traits - but not memory manager.